File: ARCADE/The Core/Level 5_2: arrayReplace.cs
    using System;
     
    public class Test
    {
    	public static int[] arrayReplace(int[] inputArray, int elemToReplace, int substitutionElem){
            for (int i = 0; i < inputArray.Length; i++){
                if (inputArray[i] == elemToReplace) inputArray[i] = substitutionElem;
            }
            return inputArray;
        }
    	public static void Main()
    	{
    		int [] inputArray = new int []{1, 2, 1};
    		int elemToReplace = 1, substitutionElem = 3;
    		var result = arrayReplace(inputArray, elemToReplace, substitutionElem);
    		foreach(int i in result){
    			Console.Write(i + " ");
    		}
    	}
    }

File: ARCADE/The Core/Level 5_2: arrayReplace.pas
program arrayReplace;
Type
    ArrayInteger = Array of integer;

function arrayReplace(inputArray: ArrayInteger; elemToReplace: integer; substitutionElem: integer): ArrayInteger;
var i :  integer;
begin
    for i := 0 to length(inputArray) - 1 do
        if(inputArray[i] = elemToReplace) then inputArray[i] := substitutionElem;
end;

var i, elemToReplace, substitutionElem : integer;
    A : ArrayInteger;
begin
	A := ArrayInteger.Create(1, 2, 1);
	elemToReplace := 1; substitutionElem := 3;
	A := arrayReplace(A, elemToReplace, substitutionElem);
	for i := 0 to length(A) - 1 do
	    write(A[i], ' ');
end.
